src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs

```diff
@@ -67,33 +70,42 @@ private static void SameExceptForBaseDir(string zipNoBaseDir, string zipBaseDir,
             }
         }
 
-        [Fact]
-        public void ExtractToDirectoryNormal()
+        [Theory]
+        [InlineData("normal.zip", "normal")]
+        [InlineData("empty.zip", "empty")]
+        [InlineData("explicitdir1.zip", "explicitdir")]
+        [InlineData("explicitdir2.zip", "explicitdir")]
+        [InlineData("appended.zip", "small")]
+        [InlineData("prepended.zip", "small")]
+        [InlineData("noexplicitdir.zip", "explicitdir")]
+        public void ExtractToDirectoryNormal(string file, string folder)
         {
-            TestExtract(zfile("normal.zip"), zfolder("normal"));
-            TestExtract(zfile("empty.zip"), zfolder("empty"));
-            TestExtract(zfile("explicitdir1.zip"), zfolder("explicitdir"));
-            TestExtract(zfile("explicitdir2.zip"), zfolder("explicitdir"));
-            TestExtract(zfile("appended.zip"), zfolder("small"));
-            TestExtract(zfile("prepended.zip"), zfolder("small"));
-            TestExtract(zfile("noexplicitdir.zip"), zfolder("explicitdir"));
+            string zipFileName = zfile(file);
+            string folderName = zfolder(folder);
+            using (var tempFolder = new TempDirectory(GetTestFilePath()))
+            {
+                ZipFile.ExtractToDirectory(zipFileName, tempFolder.Path);
+                DirsEqual(tempFolder.Path, folderName);
+            }
         }
 
         [Fact]
-        [Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)] // Jenkins fails with unicode characters [JENKINS-12610]
-        public void ExtractToDirectoryUnicode()
+        public void ExtractToDirectoryNull()
         {
-            TestExtract(zfile("unicode.zip"), zfolder("unicode"));
+            Assert.Throws<ArgumentNullException>(() => ZipFile.ExtractToDirectory(null, GetTestFilePath()));
         }
 
-        private void TestExtract(string zipFileName, string folderName)
+        [Fact]
+        public void ExtractToDirectoryUnicode()
         {
+            string zipFileName = zfile("unicode.zip");
+            string folderName = zfolder("unicode");
             using (var tempFolder = new TempDirectory(GetTestFilePath()))
             {
                 ZipFile.ExtractToDirectory(zipFileName, tempFolder.Path);
-                DirsEqual(tempFolder.Path, folderName);
-
-                Assert.Throws<ArgumentNullException>(() => ZipFile.ExtractToDirectory(null, tempFolder.Path));
+                IEnumerable<string> actual = Directory.EnumerateFileSystemEntries(tempFolder.Path, "*", SearchOption.AllDirectories);
+                IEnumerable<string> expected = Directory.EnumerateFileSystemEntries(folderName, "*", SearchOption.AllDirectories);
+                Assert.True(Enumerable.SequenceEqual(expected.Select(i => Path.GetFileName(i)), actual.Select(i => Path.GetFileName(i))));
```

Comment:
These three lines are used twice in this file. The usage earlier in the `CreateFromDirectoryUnicode` test is similar but not enough to cleanly parameterize with this one.

I'll move these to a helper method though.